docs/api-reference/layer.md

```diff
@@ -288,6 +288,21 @@ And is expected to return an array of "ranges", in the form of `{startRow, endRo
 
 **This feature is experimental and intended for advanced use cases.** Note that it only rewrites part of a buffer, not remove or insert, therefore the user of this prop is responsible of making sure that all the unchanged objects remain at the same indices between `oldData` and `newData`. This becomes trickier when dealing with data of dynamic lengths, for example `PathLayer`, `PolygonLayer` and `GeoJsonLayer`. Generally speaking, it is not recommended to use this feature when the count of vertices in the paths/polygons may change.
 
+##### `positionFormat` (String, optional)
+
+One of `'XYZ'`, `'XY'`.
+
+This prop is currently only effective in `PathLayer`, `SolidPolygonLayer` and `PolygonLayer`.
+
+Default `'XYZ'`.
+
+##### `colorFormat` (String, optional)
```

Comment:
It might simplify things to make these numbers, e.g. `positionDimensions` and `colorChannels`. The setting the size would just be `size: this.props.colorChannels` (with a default of 4). Also leaves the door open for passing fewer components (if that might ever be useful?).

Reply:
Unfortunately `positionFormat` has been a public API since v6. I can change the usage to `size: this.props.colorFormat.length` though.